saleor/graphql/webhook/subscription_types.py

```diff
@@ -38,7 +44,80 @@
 }
 
 
-class AppBase(AbstractType):
+class IssuingPrincipalType(Enum):
+    APP = "app"
+    USER = "user"
+
+
+class IssuingPrincipal(ObjectType):
+    id = graphene.String(
+        description="Issuing principal id" + ADDED_IN_34 + PREVIEW_FEATURE,
+    )
+    type = graphene.String(
+        description="Issuing principal type." + ADDED_IN_34 + PREVIEW_FEATURE,
+    )
+
+    @staticmethod
+    def resolve_id(_root, info):
+        if isinstance(info.context.requestor, (User, AnonymousUser)):
+            return graphene.Node.to_global_id("User", info.context.requestor.id)
+        return info.context.requestor.name
+
+    @staticmethod
+    def resolve_type(_root, info):
+        if isinstance(info.context.requestor, (User, AnonymousUser)):
+            return IssuingPrincipalType.USER.value
+        return IssuingPrincipalType.APP.value
+
+
+class Meta(ObjectType):
+    issued_at = graphene.DateTime(
+        description="Time of generating the event." + ADDED_IN_34 + PREVIEW_FEATURE,
+    )
+    version = graphene.String(
+        description="Saleor version." + ADDED_IN_34 + PREVIEW_FEATURE
+    )
+    issuing_principal = graphene.Field(
+        IssuingPrincipal,
+        description="Look up a issuing principal." + ADDED_IN_34 + PREVIEW_FEATURE,
+    )
+    app = graphene.Field(
+        "saleor.graphql.app.types.App",
+        description="Look up a recipient of the webhook."
+        + ADDED_IN_34
+        + PREVIEW_FEATURE,
+    )
+
+    @staticmethod
+    def resolve_issued_at(_root, _info):
+        return timezone.now()
```

Comment:
Is it correct? If we wan't to resolve older events it will be always current datetime.

Reply:
This needs to be timestamped when the event happens, not when it's delivered.

Reply:
`create_deliveries_for_subscriptions` is called when event happens and `EventPayload` is created. So `issued_at =/= delivery time` or I'm missing something?

Reply:
We want all payloads to have the exact same timestamp (and the timeline events for orders should also share the same timestamp).